src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -743,6 +783,18 @@ internal void ClearPackageLevelGrouping()
             ItemsView.GroupDescriptions.Clear();
         }
 
+        internal void AddVulnerabilitiesFiltering()
+        {
+            FilterByVulnerabilities = true;
+            ItemsView.Refresh();
```

Comment:
Did you see the filter become stale? I've got to play with this myself, but I recall only a particular scenario where this is necessary. The alternative is adding Live filtering to a list collection view:

`lcv.IsLiveFiltering = true;`
` lcv.LiveFilteringProperties.Add(your property)`

Reply:
Yeah the filter became stale, I'll try `IsLiveFiltering` property, but probably is just doing a refresh too

Reply:
No, live filtering properties are subscribed to the `CollectionChanged` event. https://github.com/dotnet/wpf/blob/main/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Data/ListCollectionView.cs#L1522

`Refresh`, however, re-creates the view. We should favor collection change over Refresh as much as possible.
https://github.com/dotnet/wpf/blob/main/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Data/CollectionView.cs#L305

The only time I believe `Refresh` should be used in a case like this, is when the filter criteria itself is changing. In this case, the criteria is static, so I believe it's unnecessary.